src/integrationTest/java/uk/gov/hmcts/darts/arm/rpo/ArpRpoApiGetMasterIndexFieldByRecordClassSchemaIntTest.java

```diff
@@ -0,0 +1,178 @@
+package uk.gov.hmcts.darts.arm.rpo;
+
+import feign.FeignException;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.hmcts.darts.arm.client.ArmRpoClient;
+import uk.gov.hmcts.darts.arm.client.model.rpo.MasterIndexFieldByRecordClassSchemaResponse;
+import uk.gov.hmcts.darts.arm.exception.ArmRpoException;
+import uk.gov.hmcts.darts.arm.helper.ArmRpoHelper;
+import uk.gov.hmcts.darts.arm.model.rpo.MasterIndexFieldByRecordClassSchema;
+import uk.gov.hmcts.darts.common.entity.ArmRpoExecutionDetailEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.enums.ArmRpoStateEnum;
+import uk.gov.hmcts.darts.common.enums.ArmRpoStatusEnum;
+import uk.gov.hmcts.darts.testutils.IntegrationBase;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+
+class ArpRpoApiGetMasterIndexFieldByRecordClassSchemaIntTest extends IntegrationBase {
```

Comment:
I considered writing integration tests like this for `getProfileEntitlements` but I didn't think there was much value, as the unit tests had already verified the vast majority of the logic.

We generally don't integration test low-level service code because integration tests are relateively expensive. Instead we favour broader integration tests invoked at the entry point of the functionality (ie. the incoming API call, or invocation of an automated task).

If we assume the functionality will be integration tested at that higher level (including what happens when each stage fails), do you think we need these tests? Or can any of these scenarios be covered by unit tests?

Reply:
Personally I think we should have integration tests as they are more realistic than unit tests which can be mocked to death. You can't and shouldn't rely on unit tests. Also each ARM RPO endpoint has its own rules which may well get overlooked when all the endpoints are joined together.

Reply:
Bear in mind we’ll have integration tests (at some point) to verify the overall flow (integrating all the endpoints), so we should try to avoid too much overlap/duplication of integration test cases. But if you feel there are scenarios here that can’t be tested at that broader level (or as a unit test) that’s fine.